cli/parser/arguments/arguments.go

```diff
@@ -0,0 +1,31 @@
+package arguments
+
+type Argument interface {
```

Comment:
could you add a comment here about what this represents and maybe also point out the two notable implementations (PositionalArgument in this package, and Option in the options package?)

Reply:
Absolutely, thanks for the feedback.